### PR TITLE
Float support

### LINKED ISSUE #64: Interpret float binops
https://github.com/dada-lang/dada/pull/56 implements execution of int binops. Floats still need to be done. Requires consideration of how to handle float eq.

### PR DESCRIPTION
**Pull request by brson**
I found myself wanting floats for dogfooding the repl.

This is using the almost unused and unmaintained [eq_float](https://docs.rs/crate/eq-float/latest/source/src/lib.rs) crate to store Eq-implementing floats in the IR. I'm happy to do something else.

Fixes #64

### CODE REVIEW

**Comment by brson** [comment]
This is missing a negative test case for "1." I'll add it later.

**Comment by brson** [comment]
I added a test that parsing floats without decimals is an error. The span pointing to the dot is not quite right, but I don't know how to fix it offhand.

File: dada_tests/interpret/ops/ops_float.dada [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    var negneginf = 0.0 - neginf1
+    if inf1 == negneginf {
+        print("inf1 == negneginf").await
+        #! OUTPUT .*
+    }
+
+    # nan != nan
+    var nan1 = 0.0 / 0.0
+    var nan2 = 0.0 / 0.0
+    if nan1 == nan2 {
```
Comment (nikomatsakis) [inline dada_tests/interpret/ops/ops_float.dada:50]: Man, I am so tempted to change this behavior and adopt IEEE total order for floats. =)

File: dada_tests/parser/float_no_decimal.dada [hunk 2]
```diff
@@ -0,0 +1,5 @@
+fn main() {
+    a = 1.
```
Comment (nikomatsakis) [inline dada_tests/parser/float_no_decimal.dada:2]: Should we permit this? People sometimes do.

**Review by nikomatsakis — CHANGES_REQUESTED** [review]
This seems good!

Can we add a test for `1_000.222_444` or something like that?

**Comment by brson** [comment]
> This seems good!
> 
> Can we add a test for `1_000.222_444` or something like that?

I've added this test. Note the syntax supports trailing underscores like `1_.2_`. This is consistent with Rust.

File: components/dada-ir/src/code/syntax.rs [hunk 3]
```diff
@@ -101,6 +101,9 @@ pub enum ExprData {
     /// `22`, `22_222`, etc
     IntegerLiteral(Word),
 
+    /// `2.2`
+    FloatLiteral(Word, Word),
```
[the remark this answers is not in the document]
Comment (brson) [reply components/dada-ir/src/code/syntax.rs:105]: I've changed the comment to be in the spirit of your other comments on these enums, but it's a bit awkward.

File: dada_tests/interpret/ops/ops_float.dada (hunk 1 above)
Comment (brson) [reply dada_tests/interpret/ops/ops_float.dada:50]: I'll open an issue.

File: dada_tests/parser/float_no_decimal.dada (hunk 2 above)
Comment (brson) [reply dada_tests/parser/float_no_decimal.dada:2]: I have no opinion.

**Review by nikomatsakis — APPROVED** [review]
Nice!

**Pull request merged**